### PR TITLE
Errors should not be wrapped - S3 Transfer Manager

### PR DESCRIPTION
**Pull request by davidh44**
Errors should not be wrapped by SdkClientException

## Motivation and Context

Partial fix to https://github.com/aws/aws-sdk-java-v2/issues/3381, for s3 Transfer Manager

## Modifications

Update handleException() to perform instance check of throwable, and handle accordingly

## Testing

Test added

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: .changes/next-release/bugfix-S3TransferManager-f1c2208.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "category": "S3 Transfer Manager", 
+    "contributor": "", 
+    "type": "bugfix", 
+    "description": "Fixed issue where errors were being wrapped by SdkClientException"
```
Comment (joviegas) [inline .changes/next-release/bugfix-S3TransferManager-f1c2208.json:5]: Can we update the title of this PR , currently it has the username in it.
Also can we be specific as in "Fixed issues in `resumeDownloadFile` API  where errors were being wrapped by SdkClientException"

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java [hunk 2]
```diff
@@ -460,10 +461,19 @@ private static void handleException(CompletableFuture<CompletedFileDownload> ret
                                         CompletableFuture<TransferProgress> progressFuture,
                                         CompletableFuture<DownloadFileRequest> newDownloadFileRequestFuture,
                                         Throwable throwable) {
-        SdkClientException exception = SdkClientException.create("Failed to resume the request", throwable);
-        returnFuture.completeExceptionally(exception);
-        progressFuture.completeExceptionally(exception);
-        newDownloadFileRequestFuture.completeExceptionally(exception);
+        Throwable cause = throwable instanceof CompletionException ? throwable.getCause() : throwable;
+
+        if (cause instanceof SdkException || cause instanceof Error) {
```
Comment (joviegas) [inline services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java:466]: Can we have a test case for cause `instanceof SdkException` check ?
Why do we have this check here ? And in what cases we get `SdkException` ? Is it fine no to communicate           "Failed to resume the request" in the exception.

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java [hunk 3]
```diff
@@ -460,10 +461,19 @@ private static void handleException(CompletableFuture<CompletedFileDownload> ret
...
+            returnFuture.completeExceptionally(cause);
+            progressFuture.completeExceptionally(cause);
+            newDownloadFileRequestFuture.completeExceptionally(cause);
+        } else {
```
Comment (joviegas) [inline services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java:470]: Also can we reduce the code to below one , What do you say  ?

```java

    private static void handleException2(CompletableFuture<CompletedFileDownload> returnFuture,
                                        CompletableFuture<TransferProgress> progressFuture,
                                        CompletableFuture<DownloadFileRequest> newDownloadFileRequestFuture,
                                        Throwable throwable) {
        
        Throwable exceptionCause = throwable instanceof CompletionException ? throwable.getCause() : throwable;
        
        Throwable propagatedException = exceptionCause instanceof SdkException || exceptionCause instanceof Error 
                                        ? exceptionCause
                                        : SdkClientException.create("Failed to resume the request", exceptionCause);

        returnFuture.completeExceptionally(propagatedException);
        progressFuture.completeExceptionally(propagatedException);
        newDownloadFileRequestFuture.completeExceptionally(propagatedException);

    }
```

File: .changes/next-release/bugfix-S3TransferManager-f1c2208.json (hunk 1 above)
Comment (davidh44) [reply .changes/next-release/bugfix-S3TransferManager-f1c2208.json]: Updated title and description

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java (hunk 3 above)
Comment (davidh44) [reply services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java]: Agreed, looks much better

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java (hunk 2 above)
Comment (davidh44) [reply services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java]: Added additional test and attached failed message to SdkException

**Pull request merged**